contrib/platform/src/com/sun/jna/platform/win32/Advapi32Util.java

```diff
@@ -1254,32 +1245,33 @@ public static Object registryGetValue(HKEY hkKey, String subKey,
             throw new Win32Exception(rc);
         }
 
-                // Buffer is intentionally allocated larger than
-                // indicated, as function adds terminating NULL char, if it is
-                // missing. WCHAR_SIZE is added, as returning string can be
-                // char[] or wchar[] depending on w32.ascii
-        Memory byteData = new Memory(lpcbData.getValue() + Native.WCHAR_SIZE);
-        byteData.clear();
+        if ((lpType.getValue() == WinNT.REG_SZ) || (lpType.getValue() == WinNT.REG_EXPAND_SZ)) {
+            NativeWin32String data = new NativeWin32String(lpcbData.getValue());
+            rc = Advapi32.INSTANCE.RegGetValue(hkKey, new NativeWin32String(subKey), new NativeWin32String(lpValueName),
+                    Advapi32.RRF_RT_ANY, lpType, data.getPointer(), lpcbData);
 
-        rc = Advapi32.INSTANCE.RegGetValue(hkKey, subKey, lpValueName,
-            Advapi32.RRF_RT_ANY, lpType, byteData, lpcbData);
+            if (rc != W32Errors.ERROR_SUCCESS) {
+                throw new Win32Exception(rc);
+            }
 
-        if (rc != W32Errors.ERROR_SUCCESS) {
-            throw new Win32Exception(rc);
-        }
+            result = data.toString();
+        } else {
+            Memory byteData = new Memory(lpcbData.getValue());
```

Comment:
You dropped the extra `Native.WCHAR_SIZE` allocation from the original, needed for the terminating NULL.